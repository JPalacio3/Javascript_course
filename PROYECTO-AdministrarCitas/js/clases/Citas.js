// Clase para las citas
class Citas {
    constructor( citas ) {
        this.citas = [];
    }

    // Agregar una nueva cita
    agregarCita( cita ) {
        this.citas = [ ...this.citas, cita ];
    }

    // Eliminar citas
    eliminarCita( id ) {
        this.citas = this.citas.filter( cita => cita.id !== id );
    }

    // Editar citas
    editarCita( citaActualizada ) {
        this.citas = this.citas.map( cita => cita.id === citaActualizada.id ? citaActualizada : cita );
    }
}

// Definir la exportaci√≥n de la clase
export default Citas;
