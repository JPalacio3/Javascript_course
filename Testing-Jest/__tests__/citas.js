import Citas from '../js/classes/Citas';

describe( 'Probar la clase de Citas', () => {

    const citas = new Citas();
    const id = Date.now()

    test( 'Agregar una nueva cita', () => {

        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Joel',
            telefono: '3339980089',
            fecha: '10-12-2020',
            hora: '10:23',
            sintomas: 'Solo duerme'
        };

        citas.agregarCita( citaObj );
        // Prueba para agregar una cita nueva:
        expect( citas ).toMatchSnapshot();
    } );

    test( 'Actualizar una cita', () => {
        const citaActualizada = {
            id: id,
            mascota: 'Nuevo Nombre',
            propietario: 'Joel',
            telefono: '3339980089',
            fecha: '10-12-2020',
            hora: '10:23',
            sintomas: 'Solo duerme'
        };

        citas.editarCita( citaActualizada );
        // Prueba para actualizar una cita:
        expect( citas ).toMatchSnapshot();
    } );

    test( 'Eliminar Citas', () => {
        citas.eliminarCita( id );

        // Prueba de eliminaci√≥n de una cita:
        expect( citas ).toMatchSnapshot();
    } )
} );
