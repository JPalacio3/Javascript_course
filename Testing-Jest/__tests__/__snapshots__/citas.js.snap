// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar una cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:23",
      "id": 1708990811210,
      "mascota": "Nuevo Nombre",
      "propietario": "Joel",
      "sintomas": "Solo duerme",
      "telefono": "3339980089",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:23",
      "id": 1708990811210,
      "mascota": "Hook",
      "propietario": "Joel",
      "sintomas": "Solo duerme",
      "telefono": "3339980089",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Citas 1`] = `
Citas {
  "citas": [],
}
`;
